File: lib/server2.py
from flask import Flask, jsonify, request, make_response
import json

app = Flask(__name__)

def load_data():
    with open('db.json', 'r', encoding='utf-8') as f:
        return json.load(f)

def save_data(data):
    with open('db.json', 'w', encoding='utf-8') as f:
        json.dump(data, f, indent=2, ensure_ascii=False)

def make_json_response(data, status=200):
    response = make_response(jsonify(data), status)
    response.headers['X-Server-ID'] = 'server2-python'
    return response


@app.route('/drinks', methods=['GET'])
def get_all_drinks():
    data = load_data()
    return make_json_response(data['drinks'])


@app.route('/drinks/<int:id>', methods=['GET'])
def get_drink(id):
    data = load_data()
    for drink in data['drinks']:
        if drink['id'] == id:
            return make_json_response(drink)
    return make_json_response({'error': 'Nápoj nenalezen'}, 404)


@app.route('/drinks', methods=['POST'])
def create_drink():
    data = load_data()
    new_drink = request.get_json()

    if not new_drink.get('name') or not new_drink.get('price'):
        return make_json_response({'error': 'Chybí povinné údaje'}, 400)

    if data['drinks']:
        new_id = max(d['id'] for d in data['drinks']) + 1
    else:
        new_id = 1

    new_drink['id'] = new_id
    data['drinks'].append(new_drink)
    save_data(data)

    return make_json_response(new_drink, 201)


@app.route('/drinks/<int:id>', methods=['PUT'])
def update_drink(id):
    data = load_data()
    update_data = request.get_json()

    for i, drink in enumerate(data['drinks']):
        if drink['id'] == id:
            update_data['id'] = id
            data['drinks'][i] = update_data
            save_data(data)
            return make_json_response(update_data)

    return make_json_response({'error': 'Nápoj nenalezen'}, 404)


if __name__ == '__main__':
    app.run(port=8002, debug=True)

File: src/server2.js
const express = require('express');
const fs = require('fs');

const app = express();
app.use(express.json());

function loadData() {
    return JSON.parse(fs.readFileSync('db.json', 'utf-8'));
}

function saveData(data) {
    fs.writeFileSync('db.json', JSON.stringify(data, null, 2));
}

app.use((req, res, next) => {
    res.setHeader('X-Server-ID', 'server2-javascript');
    next();
});

app.get('/getTickets/all', (req, res) => {
    const data = loadData();
    res.json(data.tickets);
});

app.get('/getTicket/byId/:id', (req, res) => {
    const data = loadData();
    const ticket = data.tickets.find(t => t.id === parseInt(req.params.id));
    if (!ticket) {
        return res.status(404).json({ error: 'Vstupenka nenalezena' });
    }
    res.json(ticket);
});

app.post('/createTicket', (req, res) => {
    const data = loadData();
    const newTicket = req.body;

    if (!newTicket.name || !newTicket.table) {
        return res.status(400).json({ error: 'Chybí povinné údaje' });
    }

    let newId = 1;
    if (data.tickets.length > 0) {
        newId = Math.max(...data.tickets.map(t => t.id)) + 1;
    }

    newTicket.id = newId;
    data.tickets.push(newTicket);
    saveData(data);

    res.status(201).json(newTicket);
});

app.put('/updateTicket/:id', (req, res) => {
    const data = loadData();
    const id = parseInt(req.params.id);
    const updateData = req.body;

    const index = data.tickets.findIndex(t => t.id === id);
    if (index === -1) {
        return res.status(404).json({ error: 'Vstupenka nenalezena' });
    }

    updateData.id = id;
    data.tickets[index] = updateData;
    saveData(data);

    res.json(updateData);
});


app.listen(8002, () => {
    console.log('Server 2 běží na portu 8002');
});
